File: weektwo/revise/sorting/Selection.java
package revise.sorting;

import java.util.Arrays;

public class Selection {

    public static void main(String[] args) {
        int[] arr = { 2, 1, 5, 4, 3 };
        sort(arr);
        System.out.println(Arrays.toString(arr));
    }

    private static void sort(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            int last = arr.length - i - 1;
            int max = findIndex(arr, last);
            swap(arr, last, max);
        }
    }

    private static int findIndex(int[] arr, int last) {
        int max = 0;
        for (int i = 0; i <= last; i++) {
            if (arr[max] < arr[i]) {
                max = i;
            }
        }
        return max;
    }

    private static void swap(int[] arr, int one, int two) {
        int temp = arr[one];
        arr[one] = arr[two];
        arr[two] = temp;
    }
}

File: dsaJavascript/workout/sorting/selection.js
function swap(arr, i, j) {
  let temp = arr[i];
  arr[i] = arr[j];
  arr[j] = temp;
}

function selectionSort(arr) {
  for (let i = 0; i < arr.length - 1; i++) {
    let minIndex = i;
    for (let j = i + 1; j < arr.length; j++) {
      if (arr[j] < arr[minIndex]) {
        minIndex = j;
      }
    }

    if (minIndex !== i) {
      swap(arr, i, minIndex);
    }
  }
}

let arr = [2, 35, 9, 1, 23];

selectionSort(arr);

console.log(arr);
